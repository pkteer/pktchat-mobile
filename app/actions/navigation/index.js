// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import merge from 'deepmerge';
import {Keyboard, Platform} from 'react-native';
import {Navigation} from 'react-native-navigation';

import CompassIcon from '@components/compass_icon';
import {DeviceTypes, NavigationTypes} from '@constants';
import {CHANNEL} from '@constants/screen';
import {Preferences} from '@mm-redux/constants';
import {getTheme} from '@mm-redux/selectors/entities/preferences';
import EventEmmiter from '@mm-redux/utils/event_emitter';
import EphemeralStore from '@store/ephemeral_store';
import Store from '@store/store';

Navigation.setDefaultOptions({
    layout: {
        orientation: [DeviceTypes.IS_TABLET ? 'all' : 'portrait'],
    },
});

function getThemeFromState() {
    const state = Store.redux?.getState() || {};

    return getTheme(state);
}

export function resetToChannel(passProps = {}) {
    const theme = getThemeFromState();

    EphemeralStore.clearNavigationComponents();

    const stack = {
        children: [{
            component: {
                id: CHANNEL,
                name: CHANNEL,
                passProps,
                options: {
                    layout: {
                        componentBackgroundColor: theme.centerChannelBg,
                    },
                    statusBar: {
                        visible: true,
                    },
                    topBar: {
                        visible: false,
                        height: 0,
                        background: {
                            color: theme.sidebarHeaderBg,
                        },
                        backButton: {
                            visible: false,
                            color: theme.sidebarHeaderTextColor,
                            enableMenu: false,
                        },
                    },
                },
            },
        }],
    };

    let platformStack = {stack};
    if (Platform.OS === 'android') {
        platformStack = {
            sideMenu: {
                left: {
                    component: {
                        id: 'MainSidebar',
                        name: 'MainSidebar',
                    },
                },
                center: {
                    stack,
                },
                right: {
                    component: {
                        id: 'SettingsSidebar',
                        name: 'SettingsSidebar',
                    },
                },
            },
        };
    }

    Navigation.setRoot({
        root: {
            ...platformStack,
        },
    });
}

export function resetToSelectServer(allowOtherServers) {
    const theme = Preferences.THEMES.denim;

    EphemeralStore.clearNavigationComponents();

    Navigation.setRoot({
        root: {
            stack: {
                children: [{
                    component: {
                        id: 'SelectServer',
                        name: 'SelectServer',
                        passProps: {
                            allowOtherServers,
                        },
                        options: {
                            layout: {
                                backgroundColor: theme.centerChannelBg,
                                componentBackgroundColor: theme.centerChannelBg,
                            },
                            statusBar: {
                                visible: true,
                            },
                            topBar: {
                                backButton: {
                                    color: theme.sidebarHeaderTextColor,
                                    enableMenu: false,
                                    title: '',
                                },
                                background: {
                                    color: theme.sidebarHeaderBg,
                                },
                                visible: false,
                                height: 0,
                            },
                        },
                    },
                }],
            },
        },
    });
}

export function resetToTeams(name, title, passProps = {}, options = {}) {
    const theme = getThemeFromState();
    const defaultOptions = {
        layout: {
            componentBackgroundColor: theme.centerChannelBg,
        },
        statusBar: {
            visible: true,
        },
        topBar: {
            visible: true,
            title: {
                color: theme.sidebarHeaderTextColor,
                text: title,
            },
            backButton: {
                color: theme.sidebarHeaderTextColor,
                enableMenu: false,
                title: '',
            },
            background: {
                color: theme.sidebarHeaderBg,
            },
        },
    };

    EphemeralStore.clearNavigationComponents();

    Navigation.setRoot({
        root: {
            stack: {
                children: [{
                    component: {
                        id: name,
                        name,
                        passProps,
                        options: merge(defaultOptions, options),
                    },
                }],
            },
        },
    });
}

export function goToScreen(name, title, passProps = {}, options = {}) {
    const theme = getThemeFromState();
    const componentId = EphemeralStore.getNavigationTopComponentId();
    const defaultOptions = {
        layout: {
            componentBackgroundColor: theme.centerChannelBg,
        },
        popGesture: true,
        sideMenu: {
            left: {enabled: false},
            right: {enabled: false},
        },
        topBar: {
            animate: true,
            visible: true,
            backButton: {
                color: theme.sidebarHeaderTextColor,
                enableMenu: false,
                title: '',
                testID: 'screen.back.button',
            },
            background: {
                color: theme.sidebarHeaderBg,
            },
            title: {
                color: theme.sidebarHeaderTextColor,
                text: title,
            },
        },
    };

    Navigation.push(componentId, {
        component: {
            id: name,
            name,
            passProps,
            options: merge(defaultOptions, options),
        },
    });
}

export function popTopScreen(screenId) {
    if (screenId) {
        Navigation.pop(screenId);
    } else {
        const componentId = EphemeralStore.getNavigationTopComponentId();
        Navigation.pop(componentId);
    }
}

export async function popToRoot() {
    const componentId = EphemeralStore.getNavigationTopComponentId();

    try {
        await Navigation.popToRoot(componentId);
    } catch (error) {
        // RNN returns a promise rejection if there are no screens
        // atop the root screen to pop. We'll do nothing in this case.
    }
}

export async function dismissAllModalsAndPopToRoot() {
    await dismissAllModals();
    await popToRoot();

    EventEmmiter.emit(NavigationTypes.NAVIGATION_DISMISS_AND_POP_TO_ROOT);
}

export function showModal(name, title, passProps = {}, options = {}) {
    const theme = getThemeFromState();
    const defaultOptions = {
        modalPresentationStyle: Platform.select({ios: 'pageSheet', android: 'none'}),
        layout: {
            componentBackgroundColor: theme.centerChannelBg,
        },
        statusBar: {
            visible: true,
        },
        topBar: {
            animate: true,
            visible: true,
            backButton: {
                color: theme.sidebarHeaderTextColor,
                enableMenu: false,
                title: '',
            },
            background: {
                color: theme.sidebarHeaderBg,
            },
            title: {
                color: theme.sidebarHeaderTextColor,
                text: title,
            },
            leftButtonColor: theme.sidebarHeaderTextColor,
            rightButtonColor: theme.sidebarHeaderTextColor,
        },
    };

    EphemeralStore.addNavigationModal(name);
    Navigation.showModal({
        stack: {
            children: [{
                component: {
                    id: name,
                    name,
                    passProps,
                    options: merge(defaultOptions, options),
                },
            }],
        },
    });
}

export function showModalOverCurrentContext(name, passProps = {}, options = {}) {
    const title = passProps.title || '';

    let animations;
    switch (Platform.OS) {
    case 'android':
        animations = {
            showModal: {
                waitForRender: true,
                alpha: {
                    from: 0,
                    to: 1,
                    duration: 250,
                },
            },
            dismissModal: {
                alpha: {
                    from: 1,
                    to: 0,
                    duration: 250,
                },
            },
        };
        break;
    default:
        animations = {
            showModal: {
                enter: {
                    enabled: false,
                },
                exit: {
                    enabled: false,
                },
            },
            dismissModal: {
                enter: {
                    enabled: false,
                },
                exit: {
                    enabled: false,
                },
            },
        };
        break;
    }

    const defaultOptions = {
        modalPresentationStyle: 'overCurrentContext',
        layout: {
            backgroundColor: 'transparent',
            componentBackgroundColor: 'transparent',
        },
        topBar: {
            visible: false,
            height: 0,
        },
        animations,
    };
    const mergeOptions = merge(defaultOptions, options);

    showModal(name, title, passProps, mergeOptions);
}

export function showSearchModal(initialValue = '') {
    const name = 'Search';
    const title = '';
    const passProps = {initialValue};
    const options = {
        topBar: {
            visible: false,
            height: 0,
        },
        ...Platform.select({
            ios: {
                modalPresentationStyle: 'pageSheet',
            },
        }),
    };

    showModal(name, title, passProps, options);
}

export const showAppForm = async (form, context, theme) => {
    const closeButton = await CompassIcon.getImageSource('close', 24, theme.sidebarHeaderTextColor);

    let submitButtons;
    const customSubmitButtons = form.submit_buttons && form.fields.find((f) => f.name === form.submit_buttons)?.options;

    if (!customSubmitButtons?.length) {
        submitButtons = [{
            id: 'submit-form',
            showAsAction: 'always',
            text: 'Submit',
        }];
    }

    const options = {
        topBar: {
            leftButtons: [{
                id: 'close-dialog',
                icon: closeButton,
            }],
            rightButtons: submitButtons,
        },
    };

    const passProps = {form, context};
    showModal('AppForm', form.title, passProps, options);
};

export async function dismissModal(options = {}) {
    if (!EphemeralStore.hasModalsOpened()) {
        return;
    }

    const componentId = options.componentId || EphemeralStore.getNavigationTopComponentId();

    try {
        await Navigation.dismissModal(componentId, options);
        EphemeralStore.removeNavigationModal(componentId);
    } catch (error) {
        // RNN returns a promise rejection if there is no modal to
        // dismiss. We'll do nothing in this case.
    }
}

export async function dismissAllModals(options) {
    if (!EphemeralStore.hasModalsOpened()) {
        return;
    }

    if (Platform.OS === 'ios') {
        const modals = [...EphemeralStore.navigationModalStack];
        for await (const modal of modals) {
            await Navigation.dismissModal(modal, options);
            EphemeralStore.removeNavigationModal(modal);
        }
    } else {
        await Navigation.dismissAllModals(options);
        EphemeralStore.clearNavigationModals();
    }
}

export function setButtons(componentId, buttons = {leftButtons: [], rightButtons: []}) {
    const options = {
        topBar: {
            ...buttons,
        },
    };

    mergeNavigationOptions(componentId, options);
}

export function mergeNavigationOptions(componentId, options) {
    Navigation.mergeOptions(componentId, options);
}

export function showOverlay(name, passProps, options = {}) {
    const defaultOptions = {
        layout: {
            backgroundColor: 'transparent',
            componentBackgroundColor: 'transparent',
        },
        overlay: {
            interceptTouchOutside: false,
        },
        ...Platform.select({
            android: {
                statusBar: {
                    drawBehind: true,
                },
            },
        }),
    };

    Navigation.showOverlay({
        component: {
            name,
            passProps,
            options: merge(defaultOptions, options),
        },
    });
}

export async function dismissOverlay(componentId) {
    try {
        await Navigation.dismissOverlay(componentId);
    } catch (error) {
        // RNN returns a promise rejection if there is no modal with
        // this componentId to dismiss. We'll do nothing in this case.
    }
}

export function openMainSideMenu() {
    if (Platform.OS === 'ios') {
        return;
    }

    const componentId = EphemeralStore.getNavigationTopComponentId();

    Keyboard.dismiss();
    Navigation.mergeOptions(componentId, {
        sideMenu: {
            left: {visible: true},
        },
    });
}

export function closeMainSideMenu() {
    if (Platform.OS === 'ios') {
        return;
    }

    Keyboard.dismiss();
    Navigation.mergeOptions(CHANNEL, {
        sideMenu: {
            left: {visible: false},
        },
    });
}

export function enableMainSideMenu(enabled, visible = true) {
    if (Platform.OS === 'ios') {
        return;
    }

    Navigation.mergeOptions(CHANNEL, {
        sideMenu: {
            left: {enabled, visible},
        },
    });
}

export function openSettingsSideMenu() {
    if (Platform.OS === 'ios') {
        return;
    }

    Keyboard.dismiss();
    Navigation.mergeOptions(CHANNEL, {
        sideMenu: {
            right: {visible: true},
        },
    });
}

export function closeSettingsSideMenu() {
    if (Platform.OS === 'ios') {
        return;
    }

    Keyboard.dismiss();
    Navigation.mergeOptions(CHANNEL, {
        sideMenu: {
            right: {visible: false},
        },
    });
}
