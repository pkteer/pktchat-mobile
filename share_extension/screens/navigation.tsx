// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';

import React from 'react';

import {Preferences} from '@mm-redux/constants';

import Channels from './channel_list';
import Share from './share';
import Teams from './team_list';

const theme = Preferences.THEMES.denim;

const Stack = createStackNavigator();
const defaultNavigationOptions = {
    headerStyle: {
        backgroundColor: theme.sidebarHeaderBg,
    },
    headerTitleStyle: {
        marginHorizontal: 0,
        left: 0,
        color: theme.sidebarHeaderTextColor,
    },
    headerBackTitleStyle: {
        color: theme.sidebarHeaderTextColor,
        margin: 0,
    },
    headerTintColor: theme.sidebarHeaderTextColor,
    headerTopInsetEnabled: false,
};

function RootStack() {
    return (
        <Stack.Navigator
            initialRouteName='Post'
            screenOptions={defaultNavigationOptions}
        >
            <Stack.Screen
                name='Post'
                component={Share}
            />
            <Stack.Screen
                name='Teams'
                component={Teams}
            />
            <Stack.Screen
                name='Channels'
                component={Channels}
            />
        </Stack.Navigator>
    );
}

export default function ExtensionNavigation() {
    return (
        <NavigationContainer>
            <RootStack/>
        </NavigationContainer>
    );
}
