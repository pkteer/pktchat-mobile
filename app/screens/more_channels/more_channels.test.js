// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import * as NavigationActions from '@actions/navigation';
import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import MoreChannels from './more_channels.js';

describe('MoreChannels', () => {
    const actions = {
        handleSelectChannel: jest.fn(),
        joinChannel: jest.fn(),
        getArchivedChannels: jest.fn().mockResolvedValue({data: [{id: 'id2', name: 'name2', display_name: 'display_name2', delete_at: 123}]}),
        getChannels: jest.fn().mockResolvedValue({data: [{id: 'id', name: 'name', display_name: 'display_name'}]}),
        getSharedChannels: jest.fn().mockResolvedValue({data: [{id: 'id3', name: 'shared_channel', display_name: 'shared_channel_name', shared: true}]}),
        searchChannels: jest.fn(),
        setChannelDisplayName: jest.fn(),
    };

    const baseProps = {
        actions,
        canCreateChannels: true,
        channels: [{id: 'id', name: 'name', display_name: 'display_name'}],
        sharedChannels: [{id: 'id3', name: 'shared_channel', display_name: 'shared_channel_name'}],
        archivedChannels: [{id: 'id2', name: 'archived', display_name: 'archived channel', delete_at: 123}],
        closeButton: {},
        currentUserId: 'current_user_id',
        currentTeamId: 'current_team_id',
        theme: Preferences.THEMES.denim,
        componentId: 'component-id',
        canShowArchivedChannels: true,
    };

    beforeEach(() => {
        jest.useFakeTimers();
    });

    test('should match snapshot', () => {
        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );

        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should call dismissModal on close', () => {
        const dismissModal = jest.spyOn(NavigationActions, 'dismissModal');

        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );

        wrapper.instance().close();
        expect(dismissModal).toHaveBeenCalledTimes(1);
    });

    test('should call setButtons on setHeaderButtons', () => {
        const setButtons = jest.spyOn(NavigationActions, 'setButtons');

        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );

        expect(setButtons).toHaveBeenCalledTimes(1);
        wrapper.instance().setHeaderButtons(true);
        expect(setButtons).toHaveBeenCalledTimes(2);
    });

    test('should match return value of filterChannels', () => {
        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );

        const channels = [{id: 'id', name: 'name', display_name: 'display_name'}];
        expect(wrapper.instance().filterChannels(channels, 'name')).toEqual([channels[0]]);
        expect(wrapper.instance().filterChannels(channels, 'display_name')).toEqual([channels[0]]);
        expect(wrapper.instance().filterChannels(channels, 'none')).toEqual([]);
        expect(wrapper.instance().filterChannels(channels, 'display_none')).toEqual([]);
    });

    test('should match state on cancelSearch', () => {
        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );

        wrapper.setState({term: 'term'});
        wrapper.instance().cancelSearch(true);
        expect(wrapper.state('term')).toEqual('');
        expect(wrapper.state('channels')).toEqual(baseProps.channels);
    });

    test('should search correct channels', () => {
        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );
        const instance = wrapper.instance();

        wrapper.setState({typeOfChannels: 'public'});
        instance.searchChannels('display_name');
        expect(wrapper.state('channels')).toEqual(baseProps.channels);

        wrapper.setState({typeOfChannels: 'archived'});
        instance.searchChannels('archived channel');
        expect(wrapper.state('archivedChannels')).toEqual(baseProps.archivedChannels);

        wrapper.setState({typeOfChannels: 'shared'});
        instance.searchChannels('shared');
        expect(wrapper.state('sharedChannels')).toEqual(baseProps.sharedChannels);
    });

    test('Allow load more public channels', () => {
        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );
        const instance = wrapper.instance();
        wrapper.setState({typeOfChannels: 'public'});
        instance.loadedChannels({data: ['channel-1', 'channel-2']});
        expect(instance.nextPublic).toBe(true);
    });

    test('Prevent load more public channels', () => {
        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );
        const instance = wrapper.instance();
        wrapper.setState({typeOfChannels: 'public'});
        instance.loadedChannels({data: null});
        expect(instance.nextPublic).toBe(false);

        instance.loadedChannels({data: []});
        expect(instance.nextPublic).toBe(false);
    });

    test('Allow load more archived channels', () => {
        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );
        const instance = wrapper.instance();
        wrapper.setState({typeOfChannels: 'archived'});
        instance.loadedChannels({data: ['archived-1', 'archived-2']});
        expect(instance.nextArchived).toBe(true);
    });

    test('Prevent load more archived channels', () => {
        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );
        const instance = wrapper.instance();
        wrapper.setState({typeOfChannels: 'archived'});
        instance.loadedChannels({data: null});
        expect(instance.nextArchived).toBe(false);

        instance.loadedChannels({data: []});
        expect(instance.nextArchived).toBe(false);
    });

    test('Allow load more shared channels', () => {
        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );
        const instance = wrapper.instance();
        wrapper.setState({typeOfChannels: 'shared'});
        instance.loadedChannels({data: ['shared-1', 'shared-2']});
        expect(instance.nextShared).toBe(true);
    });

    test('Prevent load more shared channels', () => {
        const wrapper = shallowWithIntl(
            <MoreChannels {...baseProps}/>,
        );
        const instance = wrapper.instance();
        wrapper.setState({typeOfChannels: 'shared'});
        instance.loadedChannels({data: null});
        expect(instance.nextShared).toBe(false);

        instance.loadedChannels({data: []});
        expect(instance.nextShared).toBe(false);
    });
});
