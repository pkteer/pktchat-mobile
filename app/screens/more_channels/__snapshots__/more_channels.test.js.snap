// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoreChannels should match snapshot 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(KeyboardLayout)>
    <Connect(StatusBar) />
    <React.Fragment>
      <View
        style={
          Object {
            "height": 38,
            "marginVertical": 5,
          }
        }
        testID="more_channels.screen"
      >
        <Search
          autoCapitalize="none"
          backArrowSize={24}
          backgroundColor="transparent"
          blurOnSubmit={false}
          cancelTitle="Cancel"
          containerHeight={40}
          deleteIconSize={20}
          editable={true}
          inputHeight={33}
          inputStyle={
            Object {
              "backgroundColor": "rgba(63,67,80,0.2)",
              "color": "#3f4350",
              "fontSize": 15,
            }
          }
          keyboardAppearance="light"
          keyboardShouldPersist={false}
          keyboardType="default"
          onBlur={[Function]}
          onCancelButtonPress={[Function]}
          onChangeText={[Function]}
          onSearchButtonPress={[Function]}
          onSelectionChange={[Function]}
          placeholder="Search"
          placeholderTextColor="rgba(63,67,80,0.5)"
          returnKeyType="search"
          searchBarRightMargin={0}
          searchIconSize={24}
          showArrow={false}
          showCancel={true}
          testID="more_channels.search_bar"
          tintColorDelete="rgba(63,67,80,0.5)"
          tintColorSearch="rgba(63,67,80,0.5)"
          titleCancelColor="#3f4350"
          value=""
        />
      </View>
      <View
        testID="more_channels.channel.dropdown"
      >
        <Text
          accessibilityRole="button"
          onPress={[Function]}
          style={
            Object {
              "color": "#3f4350",
              "fontWeight": "bold",
              "marginBottom": 10,
              "marginLeft": 10,
              "marginTop": 20,
            }
          }
          testID="more_channels.channel.dropdown.public"
        >
          Show: Public Channels
            
          <CompassIcon
            name="menu-down"
          />
        </Text>
      </View>
      <CustomList
        canRefresh={false}
        data={
          Array [
            Object {
              "display_name": "display_name",
              "id": "id",
              "name": "name",
            },
          ]
        }
        extraData={true}
        listType="flat"
        loading={true}
        loadingComponent={
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "#ffffff",
                "height": 70,
                "justifyContent": "center",
              }
            }
          >
            <InjectIntl(FormattedText)
              defaultMessage="Loading Channels..."
              id="mobile.loading_channels"
              style={
                Object {
                  "color": "rgba(63,67,80,0.6)",
                }
              }
            />
          </View>
        }
        noResults={null}
        onLoadMore={[Function]}
        onRowPress={[Function]}
        renderItem={[Function]}
        shouldRenderSeparator={true}
        showNoResults={true}
        testID="more_channels.custom_list"
        theme={
          Object {
            "awayIndicator": "#ffbc1f",
            "buttonBg": "#1c58d9",
            "buttonColor": "#ffffff",
            "centerChannelBg": "#ffffff",
            "centerChannelColor": "#3f4350",
            "codeTheme": "github",
            "dndIndicator": "#d24b4e",
            "errorTextColor": "#d24b4e",
            "linkColor": "#386fe5",
            "mentionBg": "#ffffff",
            "mentionColor": "#1e325c",
            "mentionHighlightBg": "#ffd470",
            "mentionHighlightLink": "#1b1d22",
            "newMessageSeparator": "#cc8f00",
            "onlineIndicator": "#3db887",
            "sidebarBg": "#1e325c",
            "sidebarHeaderBg": "#192a4d",
            "sidebarHeaderTextColor": "#ffffff",
            "sidebarTeamBarBg": "#14213e",
            "sidebarText": "#ffffff",
            "sidebarTextActiveBorder": "#5d89ea",
            "sidebarTextActiveColor": "#ffffff",
            "sidebarTextHoverBg": "#28427b",
            "sidebarUnreadText": "#ffffff",
            "type": "Denim",
          }
        }
      />
    </React.Fragment>
  </Connect(KeyboardLayout)>
</RNCSafeAreaView>
`;
